content/spaces/setup.textile

```diff
@@ -0,0 +1,79 @@
+languages:
+  - javascript
+---
+
+Use these instructions to install, authenticate and instantiate the Spaces SDK.
+
+h2(#authenticate). Authenticate
+
+An "API key":/auth#api-keys is required to authenticate with Ably. API keys are used either to authenticate directly with Ably using "basic authentication":/auth/basic, or to generate tokens for untrusted clients using "token authentication":/auth/token.
+
+"Sign up":https://ably.com/sign-up to Ably to create an API key in the "dashboard":https://ably.com/dashboard or use the "Control API":/account/control-api to create an API programmatically.
+
+To use the Spaces SDK, the API key requires the following "capabilities:":/auth/capabilities
+
+* Publish
+* Subscribe
+* Presence
+* History
+
+h2(#install). Install and instantiate
+
+A Spaces client establishes a connection to Ably using the Ably JavaScript SDK. Therefore the Ably JavaScript SDK is installed and instantiated alongside the Spaces SDK.
+
+Both SDKs are available as "NPM modules":#npm and via "CDN.":#cdn
+
+<aside data-type='important'>
+<p>The examples use "basic authentication":/auth/basic to demonstrate features for convenience. In your own applications, basic authentication should never be used on the client-side as it exposes your Ably API key. Instead use "token authentication.":/auth/token</p>
+</aside>
+
+h3(#npm). Using NPM
+
+Install the Ably JavaScript SDK and the Spaces SDK:
+
+```[sh]
+npm install ably @ably/spaces
+```
+
+Import the SDKs into your file:
+
+```[javascript]
+import Spaces from '@ably/spaces';
+import { Realtime } from 'ably';
+```
+
+Instantiate a realtime client using the Ably JavaScript SDK. Pass the generated client into the Spaces constructor:
+
+```[javascript]
+const client = new Realtime.Promise({key: "<API-key>", clientId: "<client-ID>"});
+const spaces = new Spaces(client);
+```
+
+A "@ClientOptions@":/api/realtime-sdk#client-options object may be passed to the Ably JavaScript SDK to further customize the connection, however at a minimum you must set your API key and provide a @ClientId@ so that the client is "identified.":/auth/identified-clients
+
+h3(#cdn). Using a CDN
+
+Reference the Ably JavaScript SDK and the Spaces SDK within the @<head>@ of your HTML file:
+
+```[html]
+<script src="https://cdn.ably.com/lib/ably.min-1.js"></script>
+<script src="https://cdn.ably.com/spaces/0.1.0/iife/index.bundle.js"></script>
+```
+
+Instantiate a realtime client using the Ably JavaScript SDK. Pass the generated client into the Spaces constructor:
+
+```[javascript]
+const client = new Realtime.Promise({key: "<API-key>", clientId: "<client-ID>"});
+const spaces = new Spaces(client);
+```
+
+A "@ClientOptions@":/api/realtime-sdk#client-options object may be passed to the Ably JavaScript SDK to further customize the connection, however at a minimum you must set your API key and provide a @ClientId@ so that the client is "identified.":/auth/identified-clients
+
```

Comment:
Do we need to repeat the instantiate section twice? Once for CDN and another for NPM? Perhaps we can have a section called "Instantiate a Spaces client", wdyt? @m-hulbert